djangoUImanager/api/urls.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# DISCLAIMER: This software is provided "as is" without any warranty,
+# express or implied, including but not limited to the warranties of
+# merchantability, fitness for a particular purpose, and non-infringement.
+#
+# In no event shall the authors or copyright holders be liable for any
+# claim, damages, or other liability, whether in an action of contract,
+# tort, or otherwise, arising from, out of, or in connection with the
+# software or the use or other dealings in the software.
+# -----------------------------------------------------------------------------
+
+# @Author  : Tek Raj Chhetri
+# @Email   : tekraj@mit.edu
+# @Web     : https://tekrajchhetri.com/
+# @File    : urls.py
+# @Software: PyCharm
+
+
+from django.urls import path, include
+from . import views
+
+
+urlpatterns = [
```

Comment:
Your API design seems to be a bit distributed in regards to URLs -- e.g. you have a separate URL entirely for GET, PATCH, POST, etc. HTTP requests

Perhaps think about the API design as:

`/api/knowledgebase/` as the root, and then, depending on what query arguments, or HTTP request, a certain action happens.

e.g.

`/api/knowledgebase?knowledge_base=all` GET request gets all knowledge bases
`api/knowledgebase/<int:id>` PATCH request updates the object with whatever your payload contains, etc. etc